mitodl/mit-open
# Add userlists page and refactor LearningResourceCardTemplate

### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open/issues/649
Part of https://github.com/mitodl/mit-open/issues/613

### Description (What does it do?)
This PR primarily adds a page at `/userlists/` that lists the currently logged-in user's `UserList` objects. A menu item was added to `UserMenu` linking to the page. The `LearningResourceCardTemplate` page component was refactored into a `CardTemplate` page component, with `LearningResourceCardTemplate inheriting it. A new page component, `UserListCardTemplate` was also added that also inherits `CardTemplate`. Hooks were also added to handle querying user lists and user list items.

### Screenshots (if appropriate):
<img width="720" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/c5a679eb-f963-42d9-8cd0-d8f8325cdd09">
<img width="149" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/43b1bbd8-24ce-458c-897c-7ab4a9616286">

### How can this be tested?
 - Make sure you're set up for running `mit-open` locally, and then spin it up on this branch
 - Create a `UserList` by visiting http://localhost:8063/api/v1/userlists/ and putting a JSON object into the raw data input with something like:
```
{
    "topics": [],
    "title": "Test UserList",
    "description": "A UserList for testing",
    "privacy_level": "private
}
```
 - Visit the home page at http://localhost:8063/
 - Log in using the drawer in the upper right
 - Once you are returned to the home page, click the user menu again and then click "User Lists"
 - Once you are at the User Lists page, you should see "Favorites" (the default `UserList`) and the `UserList` you created

### Additional Context
It's not currently possible to create a new `UserList` on this page, or edit existing `UserList`s yet. That is part of https://github.com/mitodl/mit-open/issues/613 though, and will come in a later PR. The primary purpose of this PR is to set the stage for the rest of the work.
